### DISCUSSION TITLE
Change the router query without reloading

### DISCUSSION THREAD

**Discussion by franknoel** [vercel/next.js#13291] [Help] [answered]
I'm wondering if it's possible to change the current page query without reloading it. I'm asking this because I would like to access a query param from another component.

Here's a preview of my code:

```
  const router = useRouter();

  const handleClick = (e) => {
    e.preventDefault();
    router.replace(
      {
        pathname: router.pathname,
        query: { job: title },
      },
      `${router.asPath}`
    );
  };
```

**Comment by jamesmosier** [comment, marked as answer]
It sounds like [shallow routing](https://nextjs.org/docs/routing/shallow-routing) is what you're looking for

**Reply by franknoel** [reply]
It is, but it seems like `shallow` is ignored because my page is dynamic.
There's another discussion explaining my issue: #13212
